File: Week4_Mon/WebView/JKKWebView_JS_OC/JKKWebView_JS_OC/ViewController.m
//
//  ViewController.m
//  JKKWebView_JS_OC
//
//  Created by kongyunpeng on 1/23/16.
//  Copyright © 2016 kongyunpeng. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIWebView *webView;

@end

@implementation ViewController

-(void)loadRemoteWeb{
    NSURL *url = [NSURL URLWithString:@"http://chrisrisner.com/31-Days-of-iOS--Day-14-The-UIWebView/"];
    NSURLRequest *request = [[NSURLRequest alloc] initWithURL:url];
    _webView.scalesPageToFit = YES;
    [_webView loadRequest:request];
}


-(void)loadLocalHtml{
    NSString *filePath = [[NSBundle mainBundle] pathForResource:@"LocalHtml" ofType:@"html"];
    NSString *contents = [[NSString alloc] initWithContentsOfFile:filePath encoding:NSUTF8StringEncoding error:nil];
    [_webView loadHTMLString:contents baseURL:[NSURL URLWithString:filePath]];
   
}

-(void)loadHtmlWithJS {
    NSString *filePath = [[NSBundle mainBundle] pathForResource:@"LocalHtml" ofType:@"html"];
    NSString *contents = [[NSString alloc] initWithContentsOfFile:filePath encoding:NSUTF8StringEncoding error:nil];
   
    NSString *jsString = [NSString stringWithFormat:
                          @"<script>alert(\"This is a Java Script Alert\");</script>"];
    NSString *finalContents = [NSString stringWithFormat:@"%@%@",contents,jsString];
    [_webView loadHTMLString:finalContents baseURL:[NSURL URLWithString:filePath]];
    
}
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [self loadHtmlWithJS];
   
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Week4_Mon/WebView/JKKWebView_Call_JS_Swi/JKKWebView_Call_JS_Swi/ViewController.swift
//
//  ViewController.swift
//  JKKWebView_Call_JS_Swi
//
//  Created by kongyunpeng on 1/23/16.
//  Copyright © 2016 kongyunpeng. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UIWebViewDelegate {
    @IBOutlet weak var webView: UIWebView!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        //"alert('Trigger the JS!');"
        let path = NSBundle.mainBundle().pathForResource("LocalHtml", ofType: "html")
        do {
        let html = try String(contentsOfFile: path!, encoding: NSUTF8StringEncoding)
            webView.delegate = self
            webView.loadHTMLString(html, baseURL: nil)
            
        } catch {print(error)}
        
      
        
        
    }
    func webView(webView: UIWebView, shouldStartLoadWithRequest request: NSURLRequest, navigationType: UIWebViewNavigationType) -> Bool {
        webView.stringByEvaluatingJavaScriptFromString("alert('Triger Java Script');")
        return true
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

